abis/mlibc/signal.h

```diff
@@ -133,8 +133,9 @@ struct sigaction {
 	void (*sa_sigaction)(int, siginfo_t *, void *);
 };
 
-// TODO: this struct won't work on all arches (for example aarch64) but
-// we don't have an arch specific abi folder for mlibc yet.
+#if defined(__x86_64__) || defined(__aarch64__)
+// TODO: This is wrong for AArch64.
```

Comment:
Why is this file changed? The `mlibc` ABI shouldn't be used for `riscv64-linux` anyway, right?

Reply:
1) I've moved the `ucontext_t` struct from a posix internal header to here (and also defined it in the Linux ABI folder) so that we can `ucontext_t` from different ABIs 2) I've added an `#error` below in case a RISC-V user does accidentally consume this without modification.